internal/armada/domain/job_set_filter.go

```diff
@@ -0,0 +1,28 @@
+package domain
+
+type JobState string
```

Comment:
so this is a real nit but it's not obvious to me how all this code is better than the more succinct:
```
var validSates = map[string]bool { "queued" : true,  "pending": true, "running": true}

func IsValidFilterState(s string) bool {
	_, ok := validSates[s]
	return ok
}
```

As an aside- Ideally the states would not have mixed case e.g. we'd have "RUNNING" rather then "Running"- if we did that we could normalise the case everywhere and forget about casing issues. I'm guessing that's not possible though because Armada already uses a mixed case elsewhere?